### ISSUE TITLE
Toggle dwm can’t boot in 22H2 windows 10

### ISSUE THREAD

**Issue by VartuloFN** [amitxv/PC-Tuning#124]
I used Toggle-dwm.bat in windows 10 22H2 can’t boot 
LogonUI.exe - Application Error 
Is there any fix or i should re-install windows??
![image](https://github.com/amitxv/PC-Tuning/assets/137735109/03b4f835-f1ce-4369-b8a7-60cb730f4c91)

**Comment by amitxv** [comment]
Did you run the script on a partition that you configured by following the guide? The script is not meant to be used on stock Windows or similar

**Comment by VartuloFN** [comment]
I have tried it on 1809 and on 20H2 and it works very well but not in 22H2 not worked windows cant boot

**Comment by VartuloFN** [comment]
Yea in the same partition and I followed the guide very well i am an advanced user in windows and i have tweaked so many devices but disabling dwm in 22H2 is a problem

**Comment by amitxv** [comment]
What version of 22H2 are you using to be exact? The script works for me on ``19045.3031``

![image](https://github.com/amitxv/PC-Tuning/assets/86924251/b03f6e82-8b76-4d9e-b9b3-fe7a256fe631)

**Comment by amitxv** [comment]
Any updates?

**Comment by dougg0k** [comment]
Wouldnt this or combining this, resolve any possible glitch or error problems, be on 10 and 11? https://github.com/valinet/ExplorerPatcher/discussions/167

**Comment by VartuloFN** [comment]
idk what is the build vesion for that iso that i was using cuz i had reinstalled windows and reverty bach to 1809

**Comment by amitxv** [comment]
> Wouldnt this or combining this, resolve any possible glitch or error problems, be on 10 and 11? [valinet/ExplorerPatcher#167](https://github.com/valinet/ExplorerPatcher/discussions/167)
> 
> What is the point of disabling DWM, after tweaks are done? Resource usage waste only? In Win10 and 11.
> 
> @amitxv

Issue isn't related to the explorer shell.

DWM wastes resources even with the ``Hardware: Legacy flip`` presentation mode.

**Comment by dougg0k** [comment]
> > Wouldnt this or combining this, resolve any possible glitch or error problems, be on 10 and 11? [valinet/ExplorerPatcher#167](https://github.com/valinet/ExplorerPatcher/discussions/167)
> > What is the point of disabling DWM, after tweaks are done? Resource usage waste only? In Win10 and 11.
> > @amitxv
> 
> Issue isn't related to the explorer shell.
> 
> DWM wastes resources even with the `Hardware: Legacy flip` presentation mode.

So you mean even if you have that classic look, by disabling dwm with your script, those glitchs (broken UI blink) would still happen.

**Comment by amitxv** [comment]
What glitches do you mean specifically? The shell is still completely intractable for me.

![image](https://github.com/amitxv/PC-Tuning/assets/86924251/a20ae3c7-41d4-4646-8f7f-77e327b1d687)

**Comment by amitxv** [comment]
One thing to note is that the scripts are meant to be used only after following the guide as the services and debloating steps ensures that the UWP processes that depend on DWM are not running. I would recommend using another that is made to be used on any OS if you plan to try this out.

**Comment by dougg0k** [comment]
Nah, I tried it before, possibly wrongly in the past, and I all got was a never ending blinking UI. I also seen a video recently and was happening the same thing.

**Comment by amitxv** [comment]
``HW: Legacy flip``
![image](https://github.com/amitxv/PC-Tuning/assets/86924251/0c793afa-6e2d-4305-80fa-24dc665e7d42)

``HW: Independent flip``
![image](https://github.com/amitxv/PC-Tuning/assets/86924251/46444c9e-5b80-418e-86f4-9aea731a2482)

I would just disable DWM with ``HW: Legacy flip`` since there is no reason to have it running. IIRC DWM disabled + ``HW: Independent flip`` results in ``HW: Legacy Copy to front buffer`` which is worse than flipping.

**Comment by dougg0k** [comment]
For DWM tweaks, I do these and keep dwm on low priority.

https://github.com/dougg0k/gaming_os_tweaker/blob/main/scripts/tweaks/usability.cmd#L280

**Comment by amitxv** [comment]
I wouldn't set DWM to low priority if you indeed decide to keep it enabled as processes input. It will mess with ``HW: Independent flip`` too.

**Comment by dougg0k** [comment]
It is being disabled https://github.com/dougg0k/gaming_os_tweaker/blob/main/scripts/tweaks/usability.cmd#L312C86-L312C110

**Comment by amitxv** [comment]
By disabled I meant preventing ``dwm.exe`` running completely. I'm not sure how those registry keys influence how DWM is interacted with by the OS, ``csrss.exe`` and more. You can check with [Media eXperience Analyzer](https://github.com/djdallmann/GamingPCSetup/blob/master/CONTENT/DOCS/IMAGES/MouseTester%20-%20CPU%20Processing%20Chain%20Described.png). I highly doubt those registry keys do what you think they do and probably don't affect the processing chain.

**Comment by dougg0k** [comment]
Hm, so, if it were to be working, I should expect those rising lines to processes to not be there for dwm?

**Comment by amitxv** [comment]
Depending on what you think ``EnableDwmInputProcessing`` does, sure.

**Comment by dougg0k** [comment]
Doesnt matter what I would think, just reading the command, anyone could try interpret that the command would no longer process input in dwm. 

But yeah, for a process, it could still be unaffected even if the command were disabling the input processing, it could still be linked to it, but who know what would be happening in the communication between each. 

I didnt read anything about DWM at this point, so I dont know anything to say anything.

---

I was looking for a possible Win11 solution, does your solution works with Win11?

I saw this https://github.com/Ingan121/DWMKiller

I might get a VM up and try it out.

**Comment by VartuloFN** [comment]
> `HW: Legacy flip` ![image](https://user-images.githubusercontent.com/86924251/252152410-0c793afa-6e2d-4305-80fa-24dc665e7d42.png)
> 
> `HW: Independent flip` ![image](https://user-images.githubusercontent.com/86924251/252152469-46444c9e-5b80-418e-86f4-9aea731a2482.png)
> 
> I would just disable DWM with `HW: Legacy flip` since there is no reason to have it running. IIRC DWM disabled + windowed results in `HW: Legacy Copy to front buffer` which is worse than `HW: Independent flip`. This mimics the Windows 7 behavior.

Amit, how can i till if the process or game is running in Legacy flip or independent flip??

**Comment by amitxv** [comment]
With PresentMon https://github.com/BoringBoredom/Frame-Time-Analysis

**Comment by amitxv** [comment]
> Doesnt matter what I would think, just reading the command, anyone could try interpret that the command would no longer process input in dwm, if value were 0.
> 
> But yeah, for a process, it could still be unaffected even if the command were disabling the input processing, it could still be linked to it, but who know what would be happening in the communication between each.
> 
> I didnt read anything about DWM at this point, so I dont know anything to say anything.
> 
> I was looking for a possible Win11 solution, does your solution works with Win11?
> 
> I saw this https://github.com/Ingan121/DWMKiller/tree/master/ko/Win11 - Edit: Not working straight away. Might be because it's for a single runtime usage, you disable while running the game and enable after, but if you restart, it will break a lot of things.
> 
> I might get a VM up and try it out.
> 
> You recommend disabling services tied to UWP in your guide, but your tool seems to do more than that, would you have a list or are able to get only from UWP tied services, to be disabled along dwm script?
> 
> @amitxv

The scripts are updated for Windows 11.

> You recommend disabling services tied to UWP in your guide

There is no specific recommendation for that. I just mentioned that minimal-services handily already gets the job done without having to do other steps. ShellExperienceHost needs to be disabled too.

**Comment by dougg0k** [comment]
Testing your script on Win11 with a fresh install + updates, without disabling anything. After the restart, I see the console PC welcome message and then all black screen, just to keep the info here.

**Comment by amitxv** [comment]
> If ShellExperienceHost needs to be disabled too, maybe include that in the script?

This is handled by the services script that gets created in [this step](https://github.com/amitxv/PC-Tuning/blob/main/docs/post-install.md#configure-services-and-drivers). As I said, these scripts are meant to be used with the guide only. The scripts were working for me last month but I can re-test on Windows 11 soon, will let you know. https://github.com/amitxv/PC-Tuning/blob/405e0498d64d9bfa8984d7560bbdeefbe0ab4699/bin/minimal-services.ini#L140

**Comment by amitxv** [comment]
Still works 
![image](https://github.com/amitxv/PC-Tuning/assets/86924251/f63e491c-2549-4723-85f7-7ded40fc2bc2)

**Comment by amitxv** [comment]
Still works.

![image](https://github.com/amitxv/PC-Tuning/assets/86924251/f63e491c-2549-4723-85f7-7ded40fc2bc2)

**Comment by dougg0k** [comment]
Ok, so I tried only getting rid of ShellExperienceHost together in the script, didnt work.

Then I followed the steps you said, about generating the disabling services bat script, I did and I run the script, it restarted and windows was still working, then I run the disable-dwm, same result as before. Black screen after welcome console msg or if I press window button it shows partially, still everything broken.

I am trying this in a VM state that I saved straight after installing and updating everything.

**Comment by dougg0k** [comment]
Ok, I followed the steps, but there was nothing saying about having to run services-disable with NSudo, because some permissions were denied, less were denied after running through it. Still after running dwm-disable script it didnt cause blackscreen, but there were no taskbar and right click are broken.

**Comment by amitxv** [comment]
> there was nothing saying about having to run services-disable with NSudo

It is. [Here](https://github.com/amitxv/PC-Tuning/blob/8557258225133642b13466860a33a2015065f2a5/docs/post-install.md?plain=1#L553) and [here](https://github.com/amitxv/service-list-builder/blob/e3246d6d17c768ff144fb95f8d59bb6db9681bc0/README.md?plain=1#L27).

> right click are broken

Where? On the taskbar?

**Comment by dougg0k** [comment]
Everywhere, doesnt work. There is no taskbar to be broke in there.

**Comment by amitxv** [comment]
> Is that intentional and to use an alternative like OpenShell?

Yes.

> Disabling only shellexperiencehost is not enough, it seems disabling services are in fact necessary, any idea what are those specific services?

No idea.

**Comment by dougg0k** [comment]
It's in a non-external-gpu environment, it's in a virtual machine that I am testing, as I said before.

**Comment by dougg0k** [comment]
> Still works.
> 
> ![image](https://user-images.githubusercontent.com/86924251/252327734-f63e491c-2549-4723-85f7-7ded40fc2bc2.png)

Hm, I am testing on the same build 22621, yet different results. 

Does everything works there? Do you have a full screen screenshot?

Did you do anything different other than your script?

**Comment by dougg0k** [comment]
I added a nice way to support and test different OSs and builds in my implementation. 

Pretty much just add to the rules property in each item and should be good. That way, different OSs and builds that might require different combination of files/dll/services can easily work with the script.

https://github.com/dougg0k/gaming_os_tweaker/blob/main/scripts/optional_helpers/dwm.ps1
